zkevm-circuits/src/root_circuit.rs

```diff
@@ -175,21 +275,217 @@ where
+                            // (first_chunk.ec_permu_gamma.assigned(), &gamma.assigned()),
+                            // fingerprint
+                            (
+                                first_chunk.ec_rwtable_prev_fingerprint.assigned(),
+                                &one_const,
+                            ),
+                            (
+                                first_chunk.sc_rwtable_prev_fingerprint.assigned(),
+                                &one_const,
+                            ),
+                        ]
+                        .iter()
+                        .for_each(|(a, b)| {
+                            loader
+                                .scalar_chip()
+                                .assert_equal(&mut loader.ctx_mut(), a, b)
+                                .unwrap();
+                        });
+
+                        first_chunk
+                    })
+                    .expect("error");
+
+                // constraint consistency between chunk
+                supercircuit_instances.iter().tuple_windows().for_each(
+                    |(instance_i, instance_i_plus_one)| {
+                        vec![
+                            (
+                                instance_i.chunk_index_next.assigned(),
+                                instance_i_plus_one.chunk_index.assigned(),
+                            ),
+                            (
+                                instance_i.total_chunk.assigned(),
+                                instance_i_plus_one.total_chunk.assigned(),
+                            ),
+                            (
+                                instance_i.end_rwc.assigned(),
+                                instance_i_plus_one.initial_rwc.assigned(),
+                            ),
+                            (
+                                instance_i.pi_digest_lo.assigned(),
+                                instance_i_plus_one.pi_digest_lo.assigned(),
+                            ),
+                            (
+                                instance_i.pi_digest_hi.assigned(),
+                                instance_i_plus_one.pi_digest_hi.assigned(),
+                            ),
+                            // state circuit
+                            (
+                                instance_i.sc_permu_alpha.assigned(),
+                                instance_i_plus_one.sc_permu_alpha.assigned(),
+                            ),
+                            (
+                                instance_i.sc_permu_gamma.assigned(),
+                                instance_i_plus_one.sc_permu_gamma.assigned(),
+                            ),
+                            (
+                                instance_i.sc_rwtable_row_next_fingerprint.assigned(),
+                                instance_i_plus_one
+                                    .sc_rwtable_row_prev_fingerprint
+                                    .assigned(),
```

Comment:
I am wondering whether it's meant `instance_i.sc_rwtable_row_next_fingerprint = instance_i_plus_one.sc_rwtable_row_curr_fingerprint.assigned`, so `curr` instead of `prev` in `i_plus_one`?

Reply:
Thanks for catch up this!
`instance_i` latest accumuated value will propagate to `instance_i+1`
Fix by renamed it to  `i.curr` = `{i+1}.prev` semantically and erase `next` term to avoid confused